### PR TITLE
Add event emission to seed library

### PR DESCRIPTION
**Pull request by cloudhead**
This will be used to create the seed dashboard: https://www.figma.com/file/SQadExhzMpNYFziggQ1vDh/seed-node-ui?node-id=0%3A1

### CODE REVIEW

File: seed/src/lib.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ use std::{collections::HashSet, io, net::SocketAddr, path::PathBuf, time::Durati
 
 use futures::stream::StreamExt;
 use thiserror::Error;
+use tokio::sync::mpsc as chan;
```
Comment (kim) [inline seed/src/lib.rs:26]: I know it's a lie, but I do not want any tokio in library code unless it is absolutely required. The only exception to this rule are `main` functions, integration tests, and currently `spawn` points.

**Review by kim — CHANGES_REQUESTED** [review]

File: seed/src/lib.rs (hunk 1 above)
Comment (cloudhead) [reply seed/src/lib.rs:26]: Understandable, but how should we solve this then? I can't use `std::mpsc`, can I?

Comment (cloudhead) [reply seed/src/lib.rs:26]: Otherwise I can use a trait, but that seems a bit overkill.

Comment (kim) [reply seed/src/lib.rs:26]: `futures::channel::mpsc`

File: seed/src/main.rs [hunk 2]
```diff
@@ -81,5 +81,9 @@ async fn main() {
     };
     let node = Node::new(config).unwrap();
 
-    node.run().await.unwrap();
+    let (tx, mut rx) = tokio::sync::mpsc::channel(1);
+
+    tokio::spawn(async move { while rx.recv().await.is_some() {} });
```
Comment (kim) [inline seed/src/main.rs:86]: :thinking:

Comment (cloudhead) [reply seed/src/main.rs:86]: Yeah I mean, I guess I could log the events for now. Perhaps this `main` should be moved to `examples/`

Comment (kim) [reply seed/src/main.rs:86]: > moved to examples

Sounds sensible.

What's the idea eventually? Slap some prometheuses?

Comment (cloudhead) [reply seed/src/main.rs:86]: Ok had a chat with xla, we're gonna get #340 merged, so this file will be removed anyway. I think `radicle-bins/seed` will serve as the example then.

Comment (cloudhead) [reply seed/src/main.rs]: Create a fancy dashboard: https://www.figma.com/file/SQadExhzMpNYFziggQ1vDh/seed-node-ui?node-id=0%3A1 (updated pr description)

File: seed/src/main.rs [hunk 3]
```diff
@@ -81,5 +82,13 @@ async fn main() {
     };
     let node = Node::new(config).unwrap();
 
-    node.run().await.unwrap();
+    let (tx, mut rx) = futures::channel::mpsc::channel(1);
+
+    tokio::spawn(async move {
+        while let Some(event) = rx.next().await {
+            log::debug!("{:?}", event);
```
Comment (kim) [inline seed/src/main.rs:89]: You set up `tracing` already, no need to sneak in `log` again :+1:

File: seed/src/lib.rs [hunk 4]
```diff
@@ -21,7 +21,7 @@ extern crate async_trait;
 
 use std::{collections::HashSet, io, net::SocketAddr, path::PathBuf, time::Duration, vec};
 
-use futures::stream::StreamExt;
+use futures::{channel::mpsc as chan, sink::SinkExt, stream::StreamExt};
```
Comment (xla) [inline seed/src/lib.rs:24]:
```suggestion
use futures::{channel::mpsc as chan, sink::SinkExt as _, stream::StreamExt as _};
```

Comment (xla) [reply seed/src/lib.rs:24]: Why the rename to `chan`? `mpsc::Sender<>` is way more expressive.

**Review by kim — APPROVED** [review]

File: seed/src/lib.rs (hunk 4 above)
Comment (cloudhead) [reply seed/src/lib.rs:24]: It makes it easy to change the channel provider, which tends to happen, as proven by this PR :wink:

Comment (xla) [reply seed/src/lib.rs]: jackie chan

**Review by xla — APPROVED** [review]
🐟 😶 ↘️ 🐁

File: seed/src/lib.rs (hunk 4 above)
Comment (kim) [reply seed/src/lib.rs]: What's the deal about renaming trait-only imports to `_`, actually? Is that something the idiotmatic polizei mandates?

Comment (xla) [reply seed/src/lib.rs]: From the bible: [Underscore Imports](https://doc.rust-lang.org/reference/items/use-declarations.html#underscore-imports)

Comment (cloudhead) [reply seed/src/lib.rs]: I actually like it because it also tells you it's a trait import. I generally find trait imports confusing.

**Pull request merged**